linalg/hypre.hpp

```diff
@@ -403,6 +414,13 @@ class HypreParMatrix : public Operator
    /// Returns the transpose of *this
    HypreParMatrix * Transpose() const;
 
+   /** Returns principle submatrix given by array of indices of connections
+       with relative size > \@ threshold in *this. */
+#if MFEM_HYPRE_VERSION >= 21800
+   HypreParMatrix * ExtractSubmatrix(Array<int> &indices,
+                                     double threshhold=0.0) const;
+#endif
+
```

Comment:
It's not clear to me exactly what this does. I'm not sure `ExtractSubmatrix` is the right name for this function.

Reply:
I don't follow what you mean here, this routine quite literally extracts a principle sub matrix from A and uses it to define a new parallel matrix.

Reply:
Maybe `ExtractPrincipalSubmatrix` would be better. I guess I'm still a bit confused about what this does exactly. I guess the rows/columns of the extracted submatrix are determined by the array `indices` and the threshold (just noticed typo: `threshhold` -> `threshold`), but I don't know how this works.

Reply:
Think of, e.g., multigrid and you split points into C-points and F-points. If you past in the list of F-points as "indices" for the matrix A, this returns Aff as a HypreParMatrix. In principle Hypre allows for non principle sub matrices, e.g., extract Afc (hence the more general name) but I didn't add an interface for that. I can change to PrincipleSubmatrix though to be more precise.

Reply:
Ok, that makes sense. And then entries in Aff whose magnitude is less than the threshold are dropped?